### PR TITLE
do not hardcode string length

### PR DESCRIPTION
**Pull request by N-R-K**
just declare an array and use sizeof operator on it. less error prone this way.

### CODE REVIEW

File: src/options.c [hunk 1]
```diff
@@ -536,9 +537,8 @@ static void showVersion(void)
 
 char *optionsNameThumbnail(const char *name)
 {
-    const char *const thumbSuffix = "-thumb";
-    const size_t thumbSuffixLength = 7;
-    const size_t newNameLength = strlen(name) + thumbSuffixLength;
+    const char thumbSuffix[] = "-thumb";
+    const size_t newNameLength = strlen(name) + sizeof(thumbSuffix);
     char *newName = calloc(1, newNameLength);
```
Comment (guijan) [inline src/options.c:542]: Change this calloc call to malloc too.

**Review by guijan — CHANGES_REQUESTED** [review]
Maybe this version of the whole function is better; it's definitely briefer, but I'm not sure if it's simpler:

```c
char *optionsNameThumbnail(const char *name)
{
    const char thumbSuffix[] = "-thumb";
    const size_t newNameLength = strlen(name) + sizeof(thumbSuffix);
    char *newName = malloc(newNameLength);

    if (!newName)
        err(EXIT_FAILURE, "Unable to allocate thumbnail");

    size_t basenameEnd = strcspn(name, ".");
    memcpy(newName, name, basenameEnd);
    strlcpy(newName+basenameEnd, thumbSuffix, newNameLength-basenameEnd);
    strlcat(newName, name+basenameEnd, newNameLength);

    return newName;
}
```
Up to you.

**Comment by N-R-K** [comment]
> Change this calloc call to malloc too.

Sure, seems fine.

>     size_t basenameEnd = strcspn(name, ".");

I don't believe this would work, e.g `a.b.c` - the extension here should be `.c` but this will end up selecting `.b.c` instead. `memrchr` is what you want here (widely supported, but unfortunately non-standard).

>     memcpy(newName, name, basenameEnd);
>     strlcpy(newName+basenameEnd, thumbSuffix, newNameLength-basenameEnd);
>     strlcat(newName, name+basenameEnd, newNameLength);

If we're gonna use `memcpy` then I'd go all the way and use `mempcpy` (non-standard, but takes a single line to implement `return (char *)memcpy(dst, src, n) + n;`).

But I see that this project prefers using these BSD functions - so I'd rather not bikeshed :) (My opinion of `strl{cpy,cat}` is not positive).

**Comment by guijan** [comment]
Good catch. I think the function shouldn't be rewritten now.

**Review by guijan — APPROVED** [review]

**Pull request merged**